src/lightning/fabric/strategies/fsdp.py

```diff
@@ -664,6 +673,10 @@ def _setup_distributed(self) -> None:
         self._process_group_backend = self._get_process_group_backend()
         assert self.cluster_environment is not None
         _init_dist_connection(self.cluster_environment, self._process_group_backend, timeout=self._timeout)
+        if self._process_group is None:
+            self._process_group = _get_process_group(
+                sharding_strategy=self.sharding_strategy, node_rank=self.node_rank, local_world_size=self.num_processes
+            )
```

Comment:
Why did you decide to expose the `process_group` and add this logic? Is my exploration at https://github.com/Lightning-AI/lightning/issues/18552#issuecomment-1718719047 wrong? Doesn't FSDP initialize the process group manually when HYBRID is passed with `process_group=None`?

Reply:
I decided to do it this way so that we can define the default behavior on our side. FSDP does the same but they don't document it. Here is the alternative approach: #18583